pkg/api-server/resource_endpoints.go

```diff
@@ -288,7 +291,15 @@ func (r *resourceEndpoints) validateResourceRequest(request *restful.Request, re
 	if r.descriptor.Scope == model.ScopeMesh && meshName != resourceMeta.Mesh {
 		err.AddViolation("mesh", "mesh from the URL has to be the same as in body")
 	}
-	err.AddError("", mesh.ValidateMeta(name, meshName, r.descriptor.Scope))
+	if create {
+		err.AddError("", mesh.ValidateMeta(resourceMeta, r.descriptor.Scope))
+	} else {
+		if verr, msg := mesh.ValidateMetaBackwardsCompatible(resourceMeta, r.descriptor.Scope); verr.HasViolations() {
+			err.AddError("", verr)
+		} else if msg != "" {
+			core.Log.Info(msg, "type", r.descriptor.Name, "mesh", resourceMeta.Mesh, "name", resourceMeta.Name)
```

Comment:
we should not use unnamed logs. It does not map well in Datadog. I see there is `log` in the package. Additionally, technically we should use `AddFieldsFromCtx`, but I imagine we most likely won't hit this issue with Mesh in Konnect, won't we?

Reply:
Changed this to use `log`. There isn't much we can do even if we see the message in Konnect, that's down to the user to rename their policies